validation-test/compiler_crashers_fixed/28818-hastype-e-expected-type-to-have-been-set.swift

```diff
@@ -6,5 +6,5 @@
 // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 
 // REQUIRES: asserts
-// RUN: not --crash %target-swift-frontend %s -emit-ir
+// RUN: not %target-swift-frontend %s -emit-ir
```

Comment:
Note: when you fix a crasher, please remove the `REQUIRES: asserts` and move it to `compiler_crashers_fixed/`!